arlas-core/src/main/java/io/arlas/server/core/impl/elastic/services/ElasticFluidSearch.java

```diff
@@ -971,81 +950,82 @@ private void setGeoMetricAggregationCollectField(Metric metric) throws ArlasExce
         if (order != null && on != null) {
-            if (!(aggregationBuilder instanceof GeoHashGridAggregation.Builder) && !(aggregationBuilder instanceof GeoTileGridAggregation.Builder)) {
-                SortOrder sort = SortOrder.Asc;
-                if(order.equals(Order.desc)){
-                    sort= SortOrder.Desc;
-                }
-                NamedValue<SortOrder> bucketOrder = null;
-                if (on.equals(OrderOn.field)) {
-                    bucketOrder = NamedValue.of("_key",sort);
-                } else if (on.equals(OrderOn.count)) {
-                    bucketOrder = NamedValue.of("_count",sort);
-                } else if (on.equals(OrderOn.result)) {
-                    if (metricAggregation != null) {
-                        // ORDER ON RESULT IS NOT ALLOWED ON COORDINATES (CENTROID) OR BOUNDING BOX
-                        if (!metricAggregation.split(":")[0].equalsIgnoreCase(CollectionFunction.GEOBBOX.name())
-                                && !metricAggregation.split(":")[0].equalsIgnoreCase(CollectionFunction.GEOCENTROID.name())) {
-                            bucketOrder = NamedValue.of(metricAggregation,sort);
-                        } else {
-                            throw new BadRequestException(ORDER_ON_GEO_RESULT_NOT_ALLOWED);
-                        }
-                    } else {
-                        throw new BadRequestException(ORDER_ON_RESULT_NOT_ALLOWED);
-                    }
-                }
-                if (aggregationBuilder instanceof DateHistogramAggregation.Builder ) {
-                    aggregationBuilder = ((DateHistogramAggregation.Builder) aggregationBuilder).order(bucketOrder);
-
-                }else if(aggregationBuilder instanceof HistogramAggregation.Builder){
-                    aggregationBuilder = ((HistogramAggregation.Builder) aggregationBuilder).order(bucketOrder);
+            if (isGeo) {
+                throw new NotAllowedException(ORDER_PARAM_NOT_ALLOWED);
+            }
+            NamedValue<SortOrder> bucketOrder = getSortOrderNamedValue(metricAggregation, order, on);
+            orderIfSupported(aggregationBuilder, bucketOrder);
+        } else if (order != null || on != null) {
+            if (isGeo) {
+                throw new NotAllowedException(ORDER_PARAM_NOT_ALLOWED);
+            }
+            if (order == null) {
+                throw new BadRequestException(ORDER_NOT_SPECIFIED);
+            } else {
+                throw new BadRequestException(ON_NOT_SPECIFIED);
+            }
+        }
+    }
 
-                }else if(aggregationBuilder instanceof TermsAggregation.Builder){
-                    aggregationBuilder = ((TermsAggregation.Builder) aggregationBuilder).order(bucketOrder);
+    private static <T> void orderIfSupported(ObjectBuilder<T> aggregationBuilder, NamedValue<SortOrder> bucketOrder) throws NotAllowedException {
+        if (aggregationBuilder instanceof DateHistogramAggregation.Builder builder ) {
```

Comment:
is it possible to summerize the if else blocks with || in one if block ?

Reply:
No sorry